apps/admin/frontend/src/components/discard_changes_modal.tsx

```diff
@@ -0,0 +1,34 @@
+import { Button, Modal } from '@votingworks/ui';
+import React from 'react';
+import styled from 'styled-components';
+
+const EqualWidthButton = styled(Button)`
+  flex: 1 1 0;
+`;
+
+export function DiscardChangesModal({
+  onBack,
+  onDiscard,
+}: {
+  onBack: () => void;
+  onDiscard: () => void;
+}): JSX.Element {
+  const content = 'Your adjudications for this ballot have not been saved.';
+
+  return (
+    <Modal
+      title="Unsaved Changes"
+      content={content}
+      actions={
+        <React.Fragment>
+          <EqualWidthButton variant="primary" onPress={onDiscard}>
+            Discard Changes
+          </EqualWidthButton>
+          <EqualWidthButton variant="neutral" onPress={onBack}>
```

Comment:
Trigger `onBack` with the `Esc` key?

Reply:
I haven't been able to get the listener to work for the escape key, so I've made a [ticket](https://github.com/votingworks/vxsuite/issues/6323) to fix this.